### PR TITLE
Reorganize coordinate separation method tests

### PR DESCRIPTION
**Pull request by eerovaher**
### Description

There are several `BaseCoordinateFrame` methods that compute separations between two coordinates: `separation()` and `position_angle()` give the magnitude and direction of the separation on the sky and `separation_3d()` gives the 3d-separation. Furthermore, these methods can be called from directly from a `BaseCoordinateFrame` instance or through a `SkyCoord`. Currently the tests for these methods are spread across several files. Having them in a single file is convenient for anyone working on those methods and also allows them to be parametrized over the different combinations of the involved coordinate classes. It is worth pointing out that the `BaseCoordinateFrame` and `SkyCoord` tests being separate has in fact caused problems where the `SkyCoord` method has had a bug fixed, but the analogous `BaseCoordinateFrame` method still had that bug.

Furthermore, some of the current tests do not check the methods properly. They either check that the output of the methods is in very loose bounds or, in a couple of cases, only check that the method calls do not raise any errors or warnings. The new tests are more rigorous.

- [x] By checking this box, the PR author has requested that maintainers do **NOT** use the "Squash and Merge" button. Maintainers should respect this when possible; however, the final decision is at the discretion of the maintainer that merges the PR.

### CODE REVIEW

File: astropy/coordinates/tests/test_angular_separation.py [hunk 1]
```diff
@@ -46,58 +45,15 @@ def test_angsep():
...
-    This tests if `separation` works for FK5 objects.
-
-    This is a regression test for github issue #891
-    """
-    a = FK5(1.0 * u.deg, 1.0 * u.deg)
-    b = FK5(2.0 * u.deg, 2.0 * u.deg)
-    a.separation(b)
-
-
 def test_proj_separations():
```
Comment (mhvk) [inline astropy/coordinates/tests/test_angular_separation.py:48]: Change the name? Maybe `test_cannot_add_subtract()`?

File: astropy/coordinates/tests/test_separation.py [hunk 2]
```diff
@@ -0,0 +1,346 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+
+"""
+Tests for `position_angle()`, `separation()` and `separation_3d` methods. They
```
Comment (mhvk) [inline astropy/coordinates/tests/test_separation.py:4]: `()` also for `_3d`

File: astropy/coordinates/tests/test_separation.py [hunk 3]
```diff
@@ -0,0 +1,346 @@
...
+    ],
+    ids=lambda x: x.pytest_id,
+)
+@pytest.mark.parametrize("method", ["position_angle", "separation", "separation_3d"])
+def test_equinox_conversion(coord, other_coord, method):
+    """
+    Regression test for
+        - #868, #891: methods raised errors in case of frames with equinoxes
+        - #3106, #15659: unspecified equinox should be interpreted as frame default,
+            both in `SkyCoord` methods (#3106) and in `BaseCoordinateFrame` methods with
```
Comment (mhvk) [inline astropy/coordinates/tests/test_separation.py:216]: alignment a bit off.

File: astropy/coordinates/tests/test_separation.py [hunk 4]
```diff
@@ -0,0 +1,346 @@
...
+            other_class(125 * u.deg, 0 * u.deg, 4 * u.one)
+        ),
+        5 * u.one,
+    )
+
+
+@pytest.mark.parametrize("dimensionless_class", [SkyCoord, ICRS])
+@pytest.mark.parametrize("length_class", [SkyCoord, ICRS])
+def test_separation_3d_distance_dimension_mismatch(length_class, dimensionless_class):
+    dimensionless_coord = dimensionless_class(1 * u.deg, -2 * u.deg, 14)
```
Comment (mhvk) [inline astropy/coordinates/tests/test_separation.py:243]: This feels weird - better not to put in any number, so one gets a `UnitSphericalRepresentation` - a distance without units should really lead to an error... 

EDIT: and now I see that case is tested below. I feel we should not test behaviour that is wrong, so would rather omit this test and have a separate PR that ensures we do not pass in dimensionless numbers for the distance...

File: astropy/coordinates/tests/test_separation.py [hunk 5]
```diff
@@ -0,0 +1,346 @@
...
+    ):
+        coord_with_distance.separation_3d(coord_no_distance)
+
+
+@pytest.mark.parametrize("coord_class", [SkyCoord, ICRS])
+@pytest.mark.parametrize(
+    "radial_velocity",
+    [
+        pytest.param({}, id="only_one_with_rv"),
+        pytest.param({"radial_velocity": -108 * u.km / u.s}, id="both_have_rv"),
```
Comment (mhvk) [inline astropy/coordinates/tests/test_separation.py:278]: This seems a bit weird - why check for RV but not proper motion? Maybe parametrize `kwargs` and add that? Though logically, the test cases should then have proper motion too...

**Review by mhvk — APPROVED** [review]
Thanks, that is *much* better. My comments are all tiny.

One slightly more general one: in some later tests, there is the there is either no or an unimportant difference in the coordinates - only the class is different. For these cases, readability might be served better if you just parametrize over the class and create the `SeparationExpectation` inside. But really not a big deal, so if you prefer what you have, that's totally fine.

Indeed, my comments are so nitpicky that I'll approve now.

File: astropy/coordinates/tests/test_separation.py (hunk 4 above)
Comment (eerovaher) [reply astropy/coordinates/tests/test_separation.py:243]: I didn't want to drop test cases and we do have explicit tests for this: https://github.com/astropy/astropy/blob/ea875472867f296eee3ed75989ed402d55587940/astropy/coordinates/tests/test_frames.py#L728-L732
But if you think this test really should be removed then I will.

File: astropy/coordinates/tests/test_angular_separation.py (hunk 1 above)
Comment (mhvk) [reply astropy/coordinates/tests/test_angular_separation.py:48]: Indeed, you are right that the test should be moved... While I agree the patch is large, I already reviewed it, and am happy to just have a separate commit that moves this test. But also OK to have a separate PR...

File: astropy/coordinates/tests/test_separation.py (hunk 4 above)
Comment (mhvk) [reply astropy/coordinates/tests/test_separation.py:243]: Arrgghh!  For this PR, though, more logical to keep the test indeed. It sounds like we'd need to deprecate dimensionless distance, which may not be worth it...

File: astropy/coordinates/tests/test_separation.py (hunk 5 above)
Comment (eerovaher) [reply astropy/coordinates/tests/test_separation.py:278]: I had managed to convince myself that we have existing tests that focus on radial velocity, but I can't find any such tests so I must have been mistaken. I agree the parametrization should be expanded.

**Review by mhvk — APPROVED** [review]
Super! This makes the tests a lot better organized; let's get it in!

**Pull request merged**